### PR TITLE
Single dimensional array contract

### PR DESCRIPTION
**Pull request by vivianjeng**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint --check`) has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Unirep.sol contains multidimensional array and it cannot be compiled by subgraph: https://github.com/graphprotocol/graph-cli/issues/342
Packing proofs is not really necessary and using single dimensional array can save gas when deploying

Issue Number: N/A

## What is the new behavior?

- Remove `unpackProof` function in `Unirep.sol`
- Save gas from 4251748 to 4079800 in deployment
-

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/contracts/contracts/Unirep.sol [hunk 1]
```diff
@@ -676,18 +676,12 @@ contract Unirep is IUnirep, zkSNARKHelper, Hasher, VerifySignature {
...
-        ProofsRelated memory proof;
-        // Unpack the snark proof
-        (proof.a, proof.b, proof.c) = unpackProof(input.proof);
-
         // Verify the proof
-        proof.isValid = epkValidityVerifier.verifyProof(
-            proof.a,
-            proof.b,
-            proof.c,
+        bool isValid = epkValidityVerifier.verifyProof(
```
Comment (vuvoth) [inline packages/contracts/contracts/Unirep.sol:680]: Could we return `epkValidityVerifier.verifyProof()` directly?

File: packages/contracts/contracts/Unirep.sol [hunk 2]
```diff
@@ -716,18 +710,12 @@ contract Unirep is IUnirep, zkSNARKHelper, Hasher, VerifySignature {
...
-        proof.isValid = startTransitionVerifier.verifyProof(
-            proof.a,
-            proof.b,
-            proof.c,
+        bool isValid = startTransitionVerifier.verifyProof(
+            _proof,
             publicSignals
         );
-        return proof.isValid;
+        return isValid;
```
Comment (vuvoth) [inline packages/contracts/contracts/Unirep.sol:718]: Same here

**Review by vuvoth — APPROVED** [review]
LGTM! Left few comments!

File: packages/contracts/contracts/Unirep.sol (hunk 1 above)
Comment (vivianjeng) [reply packages/contracts/contracts/Unirep.sol:680]: yeah sounds good 👍🏻

**Pull request merged**